rita_exit/src/database/database_tools.rs

```diff
@@ -654,14 +656,19 @@ fn generate_iterative_client_subnet(
     // This is the total number of client subnets available. We are checking that our iterative index
     // is lower than this number. For example, exit subnet: fd00:1000/120, client subnet /124, number of subnets will be
     // 2^(124 - 120) => 2^4 => 16
-    if ind < (1 << (subprefix - network.prefix.num)) {
-        net = net.from(&net.host_address, &net.prefix);
-        let size = net.size();
-        net.host_address += ind * size;
-        let ret = net
-            .to_string()
-            .parse()
-            .expect("Paniced while parsing exit subnet: IPAdress -> IpNetwork");
+    if ind < (1 << (subprefix - exit_sub.prefix())) {
```

Comment:
a comment about how this bitshifting works exactly would also be good.

Reply:
there is already a comment above this